### PR TITLE
feat(codes): Add auth-server verify recovery key route

### PR DESCRIPTION
**Pull request by vbudhram**
Builds off https://github.com/mozilla/fxa/pull/3957 which should be reviewed first.
Connects to #3936 

This PR adds the supporting auth-server routes to verify a recovery key before enabling it. Marking as draft until I can clean the code up a bit.

### CODE REVIEW

File: packages/fxa-auth-server/lib/routes/recovery-key.js [hunk 1]
```diff
@@ -34,35 +35,73 @@ module.exports = (log, db, Password, verifierVersion, customs, mailer) => {
...
+        if (sessionToken.tokenVerificationId) {
+          throw errors.unverifiedSession();
+        }
+
+        const { uid } = sessionToken;
+        const { recoveryKeyId } = request.payload;
+
+        // Attempt to retrieve a recovery key, if it exists and is not already enabled,
+        // then we enable it.
+        const recoveryKeyData = await db.getRecoveryKey(uid, recoveryKeyId);
```
Comment (rfk) [inline packages/fxa-auth-server/lib/routes/recovery-key.js]: A quick drive-by to say, it looks like this route would let you test whether a given `recoveryKeyId` is valid, so should probably be subject to the same rate-limiting as other routes like `GET /recoveryKey/{recoveryKeyId}`.

Comment (vbudhram) [reply packages/fxa-auth-server/lib/routes/recovery-key.js]: Ah yea that makes sense, I will make those changes. In meantime I am going to merge these changes into https://github.com/mozilla/fxa/pull/3957.
